OpenSim/Tools/InverseKinematicsTool.cpp

```diff
@@ -312,7 +312,13 @@ bool InverseKinematicsTool::run()
 
         const clock_t start = clock();
         double dt = 1.0/markersReference.getSamplingFrequency();
-        int Nframes = int((final_time-start_time)/dt)+1;
+        int Nframes = int((final_time-start_time)/dt + SimTK::SignificantReal)+1;
```

Comment:
@aseth1:

@tkuchida and I are chatting about this. If the IKTool's output is supposed to correspond to frames in the TRC file, then perhaps we should be counting the number of frames within the user-specified time range and avoid any truncation or precision issues:

```cpp
int NFrames = 0;
const auto& times = markersReference.getMarkerTable().getIndependentColumn();
int irow = 0;
while (times[irow] < start_time) {
    ++irow;
    if (irow > times.size()) throw an exception.
}
const int istart = irow;
while (times[irow] <= final_time) {
    ++irow;
    ++NFrames;
}
```

Then the loop below could set time like `s.updTime() = times[istart + i]`.

This alternative should be more consistent with the fact that `MarkersReference` does not interpolate.

Reply:
I waffled on doing that, but I agree. For IK where we are essentially processing discrete frames (and not interpolating) incrementing time is somewhat dubious. Your solution is also independent of the sampling rate and I didn't like relying on it. This is more flexible and enables varying sampling rates. I'll update.

Thanks @chrisdembia and @tkuchida!